app/assets/javascripts/components/BaseComponent.es6

```diff
@@ -11,6 +11,26 @@
       input.setAttribute('type', type);
       return input.type == type;
     }
+
+    debounce(func, wait, immediate) {
+      let timeout;
+
+      return () => {
+        const context = this,
+          args = arguments,
+          later = () => {
+            timeout = null;
+            if (!immediate) func.apply(context, args);
+          },
+          callNow = immediate && !timeout;
+
+        clearTimeout(timeout);
+
+        timeout = setTimeout(later, wait);
+
+        if (callNow) func.apply(context, args);
```

Comment:
Actually thought this wouldn't get through the linting. What do you reckon to enabling the [curly](http://eslint.org/docs/rules/curly) rule? Any opinions either way?